casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -236,43 +233,48 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
     "4! (4!)"   -> VOLATILE_EVENT             -> merges(F_, F1 | S1, S0),
     "4! (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflicts(F_, C_ | S0, S0),
     "4X (4!)"   -> VOLATILE_EVENT             -> merges(F_, F_ | S0, Nil),
-    "4X (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflictsForNow(F_, C_ | S0, Nil),
-    "C! (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflictsForNow(C_, C_ | S0, S0),
-    "CX (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflictsForNow(C_, C_ | S0, Nil),
+    "4X (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> merges(F_, C_ | S0, Nil),
+    "C! (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> conflicts(C_, C_ | S0, S0),
+    "CX (C!)"   -> PERSISTENT_COULD_NOT_MATCH -> merges(C_, C_ | S0, Nil),
     "(!4) 4!"   -> VOLATILE_EVENT             -> merges(S1 | F1, F_, S0),
     "(!4) !C"   -> VOLATILE_EVENT             -> merges(S1 | F1, S0, C0),
     "(!4) C!"   -> VOLATILE_EVENT             -> merges(S0 | F0, C_, S1),
     "(4!) CX"   -> VOLATILE_EVENT             -> merges(F_ | S0, C_, Nil),
-    "(C!) (C!)" -> PERSISTENT_COULD_NOT_MATCH -> conflictsForNow(C_ | S0, C_ | S0, Nil)
+    "(C!) (C!)" -> PERSISTENT_COULD_NOT_MATCH -> merges(C_ | S0, C_ | S0, Nil)
   )
 
-  it should "handle multi-parent blocks correctly when they operate on volatile produce/consume pairs" in effectTest {
-    mergeabilityCases.values.toList.parSequence_
+  it should "respect mergeability rules when merging blocks" in effectTest {
+    mergeabilityCases.toList
+      .map {
+        case ((name, desc), y) =>
+          val key = s"$name $desc"
+          y.attempt.map(_.fold(_ => s"$key FAILED", _ => key))
+      }
+      .parSequence
+      .map(_.filter(_.contains("FAILED")))
+      .map { r =>
```

Comment:
subsequent maps can be squashed
map after sequence is traverse

Reply:
I guess if we're just interested in failure cases (which is a greate choice!) we could just filter for Left-s

Reply:
Notice that the `desc` string is hellla long and will obscure output. I see two ways out of that:

1) make the `name`-s unique, add a validation in the test-validating test that forces that
2) add whitespace, so that the list of failures looks like:

```
$name FAILED:

$desc

$name FAILED:

$desc

...

```

Reply:
I also can't wait for the test case self-validation we've promised each other (encoding the conditions stated in `desc` strings as simple rules validating that a given example meets the preconditions), e.g.:

```
//should fail a precondition because "4" and "!" don't have same polarity
samePolarityMerge("4X !X", S0, F_, Nil)`

//should fail a precondition because S0 is not a receive as dictated by 4X
samePolarityMerge("4X 4!", S0, F_, Nil)`
```

Are you going to add it here, or in the next PR? :)

Reply:
given that I've just spotted a case where PERSISTENT_COULD_NOT_MATCH was marked as `conflicts` (should be merge for all PERSISTENT_COULD_NOT_MATCH) - having the self-validation would help reviewing this.

Reply:
I would opt for a new PR